src/IROperator.cpp

```diff
@@ -623,21 +627,9 @@ void match_types(Expr &a, Expr &b) {
 void match_bits(Expr &x, Expr &y) {
     // The signedness doesn't match, so just match the bits.
     if (x.type().bits() < y.type().bits()) {
-        Type t;
-        if (x.type().is_int()) {
```

Comment:
what the hell was going on here? Is there something special about floats?

Reply:
I don't know. The old could would have casted floats to ints though, so I doubt it was doing something useful there.